test/exprtest/blas1_axpy_copy_test.cpp

```diff
@@ -0,0 +1,144 @@
+/***************************************************************************
+ *
+ *  @license
+ *  Copyright (C) Codeplay Software Limited
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  For your convenience, a copy of the License has been included in this
+ *  repository.
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ *  SYCL-BLAS: BLAS implementation using SYCL
+ *
+ *  @filename blas1_axpy_copy_test.cpp
+ * 
+ * 	@description
+ * 			Test creates an expression tree consisting of two COPY operations
+ * 			whose resulting expressions will be fed to an AXPY operation.
+ * 			The results are validated against a CBLAS reference implementation.
+ *
+ **************************************************************************/
+#include "blas_test.hpp"
+#include "sycl_blas.hpp"
+typedef ::testing::Types<blas_test_float<>, blas_test_double<> > BlasTypes;
+
+// inputs combination
+template<typename scalar_t>
+using combination_t = std::tuple<int, scalar_t, int, int>;
+
+template <typename scalar_t>
+void run_test(const combination_t<scalar_t> combi) {
+  using type_t = blas_test_args<scalar_t, void>;
+  using blas_test_t = BLAS_Test<type_t>;
+  using executor_t = typename type_t::executor_t;
+
+  int size;
+  scalar_t alpha;
+  int incX;
+  int incY;
+  std::tie(size, alpha, incX, incY) = combi;
+
+  // Dimensions of input vectors x and y
+  int x_dim = size * incX;
+  int y_dim = size * incY;
+
+  // Input vectors x and y (y is also output of AXPY)
+  std::vector<scalar_t> v_x(x_dim);
+  std::vector<scalar_t> v_y(y_dim);
+  blas_test::utils::fill_random(v_x);
+  blas_test::utils::fill_random(v_y);
+
+  // Copy of vector x
+  std::vector<scalar_t> v_xcopy(x_dim);
+
+	// Copy of vector y
+  // Used to store output on host for later comparison with SYCL result
+	std::vector<scalar_t> v_cpu_y(y_dim);
+  
+  // Reference BLAS implementation
+  reference_blas::copy(size, v_x.data(), incX, v_xcopy.data(), incX);
+	reference_blas::copy(y_dim, v_y.data(), 1, v_cpu_y.data(), 1);
```

Comment:
As a convention, we use *Clang Format* for this project, with the configuration defined in `.clang-format`. Can you run `clang-format -i <your-files>`? I spotted a few formatting issues (e.g indent)

Reply:
I don't see the bad indents in my editor. I think it might be a tab size thing in VSCode. I'll run the format